framework/Objects/Meta_Fields.php

```diff
@@ -0,0 +1,266 @@
+<?php
+
+namespace JustCoded\WP\Framework\Objects;
+
+/**
+ * Meta_Fields is the base class for acquiring data from custom fields.
+ */
+
+class Meta_Fields {
+
+	/**
+	 * Plugin used for extending of post custom fields.
+	 *
+	 * Available values:
+	 *      'just-custom-fields'
+	 *      'advanced-custom-fields'
+	 *
+	 * @var string
+	 */
+	public $custom_fields_plugin = 'just-custom-fields';
+
+	/**
+	 * Current $post object
+	 *
+	 * @var \WP_Post
+	 */
+	public $post;
+
+	/**
+	 * Current $term object
+	 *
+	 * @var \WP_Term
+	 */
+	public $term;
+
+	/**
+	 * Returns the value of an object property.
+	 *
+	 * Do not call this method directly as it is a PHP magic method that
+	 * will be implicitly called when executing `$value = $object->property;`.
+	 *
+	 * @param string $name The property name.
+	 *
+	 * @return mixed the property value
+	 * @throws \Exception Property is not defined.
+	 * @see __set()
+	 */
+	public function __get( $name ) {
+		if ( $this->post ) {
+			// getter for magic property $this->field_*.
+			return $this->get_field( $name );
+		} else {
+			return $this->get_term_field( $name );
+		}
+
+		// trying to find get_{property} method.
+		$getter = 'get_' . $name;
+		if ( method_exists( $this, $getter ) ) {
+			return $this->$getter();
+		} else {
+			throw new \Exception( 'Getting unknown property: ' . get_class( $this ) . '::' . $name );
+		}
+	}
+
+	/**
+	 * Sets value of an object property.
+	 *
+	 * Do not call this method directly as it is a PHP magic method that
+	 * will be implicitly called when executing `$object->property = $value;`.
+	 *
+	 * @param string $name The property name or the event name.
+	 * @param mixed  $value The property value.
+	 *
+	 * @throws \Exception Property is not defined or read-only.
+	 * @see __get()
+	 */
+	public function __set( $name, $value ) {
+		$setter = 'set_' . $name;
+		if ( method_exists( $this, $setter ) ) {
+			$this->$setter( $value );
+		} elseif ( method_exists( $this, 'get_' . $name ) ) {
+			throw new \Exception( 'Setting read-only property: ' . get_class( $this ) . '::' . $name );
+		} else {
+			throw new \Exception( 'Setting unknown property: ' . get_class( $this ) . '::' . $name );
+		}
+	}
+
+	/**
+	 * Checks if the named property is set (not null).
+	 *
+	 * Do not call this method directly as it is a PHP magic method that
+	 * will be implicitly called when executing `isset($object->property)`.
+	 *
+	 * Note that if the property is not defined, false will be returned.
+	 *
+	 * @param string $name The property name or the event name.
+	 *
+	 * @return boolean Whether the named property is set (not null).
+	 */
+	public function __isset( $name ) {
+		$getter = 'get_' . $name;
+		if ( method_exists( $this, $getter ) ) {
+			return $this->$getter() !== null;
+		} else {
+			return false;
+		}
+	}
```

Comment:
please update isset to match correct data - we need to return is such meta key exists (also based on plugin type)